module.d.ts

```diff
@@ -0,0 +1,200 @@
+				/** the error thrown by the module before the error handler tried to handle it. */
+				originalError: Error;
+		  }
+		| {
+				type: "accept-errored";
+				/** The module in question. */
+				moduleId: number;
+				/** the module id owning the accept handler. */
+				dependencyId: number;
+				/** the thrown error */
+				error: Error;
+		  }
+		| {
+				type: "self-accept-errored";
+				/** The module in question. */
+				moduleId: number;
+				/** the thrown error */
+				error: Error;
+		  };
+
+	interface ApplyOptions {
+		ignoreUnaccepted?: boolean;
+		ignoreDeclined?: boolean;
+		ignoreErrored?: boolean;
+		onDeclined?(callback: (info: HotEvent) => void): void;
+		onUnaccepted?(callback: (info: HotEvent) => void): void;
+		onAccepted?(callback: (info: HotEvent) => void): void;
+		onDisposed?(callback: (info: HotEvent) => void): void;
+		onErrored?(callback: (info: HotEvent) => void): void;
+	}
+
+	const enum HotUpdateStatus {
+		idle = "idle",
+		check = "check",
+		prepare = "prepare",
+		ready = "ready",
+		dispose = "dispose",
+		apply = "apply",
+		abort = "abort",
+		fail = "fail"
+	}
+
+	interface Hot {
+		accept: {
+			(
+				modules: string | string[],
+				callback?: (outdatedDependencies: string[]) => void,
+				errorHandler?: (
+					err: Error,
+					context: { moduleId: string | number; dependencyId: string | number }
+				) => void
+			): void;
+			(
+				errorHandler: (
+					err: Error,
+					ids: { moduleId: string | number; module: NodeJS.Module }
+				) => void
+			): void;
+		};
+		status(): HotUpdateStatus;
+		decline(module?: string | string[]): void;
+		dispose(callback: (data: object) => void): void;
+		addDisposeHandler(callback: (data: object) => void): void;
+		removeDisposeHandler(callback: (data: object) => void): void;
+		invalidate(): void;
+		addStatusHandler(callback: (status: HotUpdateStatus) => void): void;
+		removeStatusHandler(callback: (status: HotUpdateStatus) => void): void;
+		data: object;
+		check(
+			autoApply?: boolean | ApplyOptions
+		): Promise<(string | number)[] | null>;
+		apply(options?: ApplyOptions): Promise<(string | number)[] | null>;
+	}
+
+	interface ExportInfo {
+		used: boolean;
+		provideInfo: boolean | null | undefined;
+		useInfo: boolean | null | undefined;
+	}
+
+	interface ExportsInfo {
+		[k: string]: ExportInfo & ExportsInfo;
+	}
+
+	interface Context {
+		resolve(dependency: string): string | number;
+		keys(): Array<string>;
+		id: string | number;
+		(dependency: string): unknown;
+	}
+}
+
+interface ImportMeta {
+	url: string;
+	webpack: number;
+	webpackHot: webpack.Hot;
+}
+
+declare const __resourceQuery: string;
+declare var __webpack_public_path__: string;
+declare var __webpack_nonce__: string;
+declare const __webpack_chunkname__: string;
+declare var __webpack_base_uri__: string;
+declare var __webpack_runtime_id__: string;
+declare const __webpack_hash__: string;
+declare const __webpack_modules__: Record<string | number, NodeJS.Module>;
```

Comment:
will not work without namespace `NodeJS`

Reply:
hm.. technically we define `NodeJS.Module`

Reply:
Maybe we can copy some properties from the NodeJs.Module into our module so it's not that empty without NodeJs types...